react-catalog/my-app/src/Details.tsx

```diff
@@ -1,29 +1,32 @@
 import { useNavigate, useParams } from 'react-router-dom';
 import { useEffect, useState } from 'react';
-import { Product, readCatalog } from './lib';
+import { type Product, readProduct, toDollars } from './lib';
 
 export function Details() {
-  const [items, setItems] = useState<Product[]>([]);
+  const [item, setItem] = useState<Product>();
   const [isLoading, setIsLoading] = useState(true);
   const [error, setError] = useState<unknown>();
   const { itemId } = useParams();
   const navigate = useNavigate();
 
   useEffect(() => {
-    async function loadItems(): Promise<void> {
+    async function loadItem(itemId: number): Promise<void> {
       try {
-        const values = await readCatalog();
-        setItems(values);
-        console.log(values);
+        const value = await readProduct(itemId);
+        setItem(value);
+        console.log(value);
       } catch (error) {
         setError(error);
         console.error('Error: ', error);
       } finally {
         setIsLoading(false);
       }
     }
-    loadItems();
-  }, []);
+    if (itemId) {
+      setIsLoading(false);
```

Comment:
This is backwards. You want to have isLoading be true while the async call is taking place. So this should be `setIsLoading(true);`.